forex.js

```diff
@@ -0,0 +1,89 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const rates = {
+        USD: { CAD: 1.25, AUD: 1.30, CNY: 6.50 },
+        CAD: { USD: 0.80, AUD: 1.04, CNY: 5.20 },
+        AUD: { USD: 0.77, CAD: 0.96, CNY: 4.95 },
+    };
+
+    function displayRates() {
+        let table = "<table><tr><th>From</th><th>To</th><th>Rate</th></tr>";
+        for (let fromCurrency in rates) {
+            for (let toCurrency in rates[fromCurrency]) {
+                table += `<tr><td>${fromCurrency}</td><td>${toCurrency}</td><td>${rates[fromCurrency][toCurrency]}</td></tr>`;
+            }
+        }
+        table += "</table>";
+        document.getElementById("currentRates").innerHTML = table;
+    }
+
+    document.getElementById('add-rate-form').addEventListener('submit', (event) => {
+        event.preventDefault();
+        const fromCurrency = document.getElementById('fromCurrency').value.toUpperCase();
```

Comment:
Controlling the value with `toUpperCase` is excellent. You should use `trim` as well to avoid such cases with spaces around the currency 
<img width="181" alt="image" src="https://github.com/user-attachments/assets/5271e251-937b-4806-8d3d-1d11e3956170">

This comment is, of course, valid for the other places where you retrieve an input value.